src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -361,7 +361,8 @@ public class GenerateProtoTask extends DefaultTask {
     Preconditions.checkState(state == State.FINALIZED, 'doneConfig() has not been called')
 
     ToolsLocator tools = project.protobuf.tools
-    Set<File> protoFiles = inputs.sourceFiles.files
+    // Sort to make sure files are in a consistent order
+    List<File> protoFiles = inputs.sourceFiles.files.sort()
```

Comment:
It's the descriptor files that can change. In the example I know of, the descriptors were being packaged into a jar which (sometimes) made the jar different.

Here's a self contained example:
```
#!/bin/bash 

declare A="a.proto"
declare B="b.proto"

cat << EOF > $A
syntax = "proto2";

package example;

message A {
    required string foo = 1;
}
EOF

cat << EOF > $B
syntax = "proto2";

package example;

message B {
    required int32 bar = 1;
}
EOF

declare FIRST_ORDER="$A $B"
declare SECOND_ORDER="$B $A"

protoc --descriptor_set_out=firstOrder.dsc $FIRST_ORDER 
protoc --descriptor_set_out=secondOrder.dsc $SECOND_ORDER
protoc --descriptor_set_out=firstOrderAgain.dsc $FIRST_ORDER
protoc --descriptor_set_out=secondOrderAgain.dsc $SECOND_ORDER

md5 firstOrder.dsc firstOrderAgain.dsc 
md5 secondOrder.dsc secondOrderAgain.dsc 
```

This should produce something like:
MD5 (firstOrder.dsc) = 358eb874091e607d4a9bc2e3d1d40caa
MD5 (firstOrderAgain.dsc) = 358eb874091e607d4a9bc2e3d1d40caa
MD5 (secondOrder.dsc) = 2c3b519a765702851c80ad65b5034381
MD5 (secondOrderAgain.dsc) = 2c3b519a765702851c80ad65b5034381

The two "first" files will be the same and the two "second" files will be the same.  I ran it twice to show that it's not something like the timestamp that makes the files different.  Looking at the generated descriptors, it seems like the only difference is the order that protoc saw the input files.